WalletWasabi.Gui/Controls/LockScreen/LockScreen.xaml.cs

```diff
@@ -0,0 +1,98 @@
+        private ContentControl LockScreenHost;
+        private LockScreenBase CurrentLockScreen;
+        private CompositeDisposable ScreenImplDisposables;
+		
+        public static readonly DirectProperty<LockScreen, LockScreenType> ActiveLockScreenTypeProperty =
+            AvaloniaProperty.RegisterDirect<LockScreen, LockScreenType>(nameof(ActiveLockScreenType),
+                                                                        o => o.ActiveLockScreenType,
+                                                                        (o, v) => o.ActiveLockScreenType = v);
+        public LockScreenType ActiveLockScreenType
+        {
+            get => _activeLockScreenType;
+            set => this.SetAndRaise(ActiveLockScreenTypeProperty, ref _activeLockScreenType, value);
+        }
+
+        public static readonly DirectProperty<LockScreen, bool> IsLockedProperty =
+            AvaloniaProperty.RegisterDirect<LockScreen, bool>(nameof(IsLocked),
+                                                              o => o.IsLocked,
+                                                              (o, v) => o.IsLocked = v);
+        public bool IsLocked
+        {
+            get => _isLocked;
+            set => this.SetAndRaise(IsLockedProperty, ref _isLocked, value);
+        }
+
+        public static readonly DirectProperty<LockScreen, string> PINHashProperty =
+            AvaloniaProperty.RegisterDirect<LockScreen, string>(nameof(PINHash),
+                                                              o => o.PINHash,
+                                                              (o, v) => o.PINHash = v);
+        public string PINHash
+        {
+            get => _pinHash;
+            set => this.SetAndRaise(PINHashProperty, ref _pinHash, value);
+        }
+
+        public LockScreen()
+        {
+            InitializeComponent();
+
+            this.LockScreenHost = this.FindControl<ContentControl>("LockScreenHost");
+
+            this.WhenAnyValue(x => x.ActiveLockScreenType)
+                .Subscribe(OnActiveLockScreenTypeChanged);
+        }
+
+        private void OnActiveLockScreenTypeChanged(LockScreenType obj)
+        {
+            ScreenImplDisposables?.Dispose();
+            ScreenImplDisposables = new CompositeDisposable();
+
+            switch (obj)
+            {
+                case LockScreenType.PINLock:
+                    var pinLock = new PINLock();
+                    CurrentLockScreen = pinLock;
+
+                    this.WhenAnyValue(x => x.PINHash)
+                        .Subscribe(y => pinLock.PINHash = y)
+						.DisposeWith(ScreenImplDisposables);
+
+                    break;
+                default:
+                    CurrentLockScreen = new SlideLock();
+                    break;
+            }
+
+            LockScreenHost.Content = CurrentLockScreen;
+
+            this.WhenAnyValue(x => x.IsLocked)
+                .BindTo(CurrentLockScreen, y => y.IsLocked)
```

Comment:
observeon uithread?

Reply:
Also ObservableAsPropertyHelpers Over Properties is not doable? https://github.com/zkSNACKs/WalletWasabi/blob/master/CONTRIBUTING.md#observableaspropertyhelpers-over-properties

Reply:
@nopara73 I've tried to implement OAPH on it but `ToProperty` gets utterly confused on `UserControl`-based classes..

Reply:
@nopara73 Id like us to have a discussion about ObserveOn UIThread, @lontivero @molnard 
perhaps at next meeting.